app/service/data_svc.py

```diff
@@ -389,15 +426,39 @@ def __init__(self, dao):
                                                           info['cleanup'].strip().encode(
                                                               'utf-8')).decode() if info.get(
                                                           'cleanup') else None,
-                                                      payload=info.get('payload'), parser=info.get('parser'))
+                                                      payload=info.get('payload'), parser=info.get('parser'),
+                                                      relationships=fact_relationships)
                 await self._delete_stale_abilities(ab)
 
-    async def _save_ability_extras(self, identifier, payload, parser):
+    async def _relationships_from_yaml(self, ability):
+        relationships = []
+        if ability.get('requires'):
+            for r in ability.get('requires'):
+                line = await self._cleanse_facts(r)
+                relationships.append(dict(property1=line[0], relationship=line[1], property2=line[2],
+                                          relationship_type='requires'))
+        if ability.get('consequence'):
+            for r in ability.get('consequence'):
+                line = await self._cleanse_facts(r)
+                relationships.append(dict(property1=line[0], relationship=line[1], property2=line[2],
+                                          relationship_type='creates'))
+        return relationships
+
+    @staticmethod
+    async def _cleanse_facts(fact):
+        f = fact.translate({ord(i): None for i in '{}#'})
+        return [i.strip() for i in f.split(',')]
+
+    async def _save_ability_extras(self, identifier, payload, parser, relationships):
         if payload:
             await self.dao.create('core_payload', dict(ability=identifier, payload=payload))
         if parser:
             parser['ability'] = identifier
             await self.dao.create('core_parser', parser)
+        if relationships:
```

Comment:
you can remove the 'if' clause here. the for-loop will not enter if relationships is empty.

Reply:
@privateducky I put the 'if' statement there to avoid the 'NoneType is not iterable' error. I'm thinking we remove the 'if' clause and have the relationships variable default to an empty list instead of None. Does that work for you?